scipy/special/add_newdocs.py

```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
     hyp2f1(a, b, c, z)
 
     Gauss hypergeometric function 2F1(a, b; c; z).
+
+    Parameters
+    ----------
+    a : double argument.
+
+    b : double argument.
+
+    c : double argument.
+
+    z : double argument.
+
+    Returns
+    -------
+    y : double argument
+        y being the solution to the second-order linear ordinary differential equation.
+
+    Notes
+    -----
+
+    .. math::
+
+        \mathrm{hyp2f1}(a, b, c, z) = 1 + \sum_{k=0}^inf. \frac {{{a+k}\choose{a}} {{b+k}}\choose{b}}}{{c+k}\choose{c}} z^(k+1).
+    
+    Cases addressed are
```

Comment:
While I am pro discussing the algorithm in the notes, you need more description here for the user to be able to understand what's happening. In particular links to the DLMF (http://dlmf.nist.gov/) are very helpful. Also this appears to only discuss what happens for the real case.